cmake/MPIConfig.cmake

```diff
@@ -0,0 +1,229 @@
+					
+					if("${PATH_EXTENSION}" STREQUAL "")
+						set(IS_WL_DIRECTORY TRUE)
+					endif()
+				endif()
+			elseif("${FIRST_ELEMENT}" MATCHES "-Wl,-rpath.*")
+				set(IS_WL_RPATH TRUE)
+			endif()
+			
+			if(IS_XLINKER_DIRECTORY)
+				math(EXPR INDEX "${INDEX} + 2")
+				message(STATUS "Found and removed RPATH control flags from MPI flags")
+			elseif(IS_WL_DIRECTORY)
+				math(EXPR INDEX "${INDEX} + 1")
+				message(STATUS "Found and removed RPATH control flags from MPI flags")
+			elseif(IS_WL_RPATH)
+				math(EXPR INDEX "${INDEX} + 1")
+				message(STATUS "Found and removed RPATH control flags from MPI flags")
+			else()
+				list(APPEND TEMP_REBUILT_LINK_OPTIONS ${FIRST_ELEMENT})
+				math(EXPR INDEX "${INDEX} + 1")
+			endif()
+		endwhile()
+		
+		set(MPI_${LANG}_LINK_OPTIONS ${TEMP_REBUILT_LINK_OPTIONS})
+					
+	endforeach()
+			
+	# create imported targets
+	# --------------------------------------------------------------------
+	foreach(LANG ${ENABLED_LANGUAGES})
+		string(TOLOWER ${LANG} LANG_LOWERCASE)
+		
+		import_libraries(mpi_${LANG_LOWERCASE} LIBRARIES ${MPI_${LANG}_LINK_OPTIONS} ${MPI_${LANG}_LIBRARIES} INCLUDES ${MPI_${LANG}_INCLUDE_PATH})
+		set_property(TARGET mpi_${LANG_LOWERCASE} PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${MPI_${LANG}_INCLUDE_PATH})
+		
+		if(MCPAR_WORKAROUND_ENABLED)
+			# use generator expression
+			set_property(TARGET mpi_${LANG_LOWERCASE} PROPERTY INTERFACE_COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:${LANG}>:${MPI_${LANG}_COMPILE_OPTIONS}>)
+		else()
+			set_property(TARGET mpi_${LANG_LOWERCASE} PROPERTY INTERFACE_COMPILE_OPTIONS ${MPI_${LANG}_COMPILE_OPTIONS})
+		endif()
+		
+		# C++ MPI doesn't like having "MPI" defined, but it's what Amber uses as the MPI switch in most programs (though not EMIL)
+		if(NOT ${LANG} STREQUAL CXX)
+			set_property(TARGET mpi_${LANG_LOWERCASE} PROPERTY INTERFACE_COMPILE_DEFINITIONS MPI)	
+		endif()
+			
+	endforeach()
+	
+	
+	# Add MPI support to an object library
+	macro(mpi_object_library TARGET LANGUAGE)
+		if(MCPAR_WORKAROUND_ENABLED)
+			# use generator expression
+			set_property(TARGET ${TARGET} APPEND PROPERTY COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:${LANGUAGE}>:${MPI_${LANGUAGE}_COMPILE_OPTIONS}>)
+		else()
+			set_property(TARGET ${TARGET} APPEND PROPERTY COMPILE_OPTIONS ${MPI_${LANGUAGE}_COMPILE_OPTIONS})
+		endif()
+
+		target_include_directories(${TARGET} PUBLIC ${MPI_${LANGUAGE}_INCLUDE_PATH})
+		
+		if(NOT ${LANGUAGE} STREQUAL CXX)
+			target_compile_definitions(${TARGET} PRIVATE MPI)
+		endif()
+	endmacro()
+
+	# make a MPI version of the thing passed 
+	# also allows switching out sources if needed
+	# INSTALL - causes the new target to get installed in the MPI component to the default location (BINDIR etc)
+	# usage: make_mpi_version(<target> <new name> LANGUAGES <language 1> [<language 2...>] [SWAP_SOURCES <source 1...> TO <replacement source 1...>] INSTALL)
+	function(make_mpi_version TARGET NEW_NAME) 
+	
```

Comment:
Can this function be accessed outside this module? I know a macro can do it but not sure about functions.

Reply:
Yep it can be accessed anywhere.